components/omega/OmegaBuild.cmake

```diff
@@ -606,14 +606,24 @@ macro(update_variables)
     set(CMAKE_INSTALL_PREFIX ${OMEGA_INSTALL_PREFIX})
   endif()
 
+  if(NOT DEFINED OMEGA_MPI_ON_DEVICE)
+    option(OMEGA_MPI_ON_DEVICE "Allow device buffers in MPI communication (default ON)." ON)
+  endif()
+
   if("${OMEGA_ARCH}" STREQUAL "CUDA")
     option(Kokkos_ENABLE_CUDA "" ON)
     option(Kokkos_ENABLE_CUDA_LAMBDA "" ON)
     add_definitions(-DOMEGA_TARGET_DEVICE)
+    if(OMEGA_MPI_ON_DEVICE AND OMEGA_CIME_MACHINE STREQUAL "pm-gpu")
+      option(Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC "" OFF)
+    endif()
```

Comment:
See also: https://kokkos.org/kokkos-core-wiki/known-issues.html?highlight=cuda_malloc#cuda. In Kokkos 4.5 they reverted that change.

Reply:
I don't know that there's any downside to have MPICH_GPU_SUPPORT_ENABLED=1, if we are not explicitly using GPU aware MPI, it has been on by default in the CIME configs this whole time we've been developing Omega anyways.

Reply:
Probably better to just remove the machine check from the `if` statement so we set `Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC=OFF` for all CUDA machines since Kokkos is rolling back the default.